File: src/main/java/com/hazza/leetcode/SegmentOne/No7ReverseInteger.java
package com.hazza.leetcode.SegmentOne;

/**
 * https://leetcode.com/problems/reverse-integer/?tab=Description#/description
 */
public class No7ReverseInteger {
    public int reverse(int x) {
        int res = 0;
        while (x != 0) {
            int tail = x % 10;
            int newRes = res * 10 + tail;
            if ((newRes - tail) / 10 != res) {
                return 0;
            }
            res = newRes;
            x = x / 10;
        }
        return res;
    }
}

File: src/main/python/com.hazza.LeetCodeByPython/SegmentOne/No7_Reverse_Integer.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Created by HazzaCheng on 2019/4/7 21:06
# https://leetcode.com/problems/reverse-integer/


class Solution:
    def reverse(self, x: int) -> int:
        sign = [1, -1][x < 0]
        res, x = 0, abs(x)
        while x != 0:
            x, tail = divmod(x, 10)
            newRes = int(res * 10 + tail)
            res = newRes

        return sign * res if -pow(2, 31) <= sign * res <= pow(2, 31)-1 else 0


if __name__ == '__main__':
    solution = Solution()
    assert solution.reverse(123) == 321
    assert solution.reverse(120) == 21
    assert solution.reverse(-123) == -321
    assert solution.reverse(1534236469) == 0